packages/common/src/utils/intent.ts

```diff
@@ -47,3 +49,137 @@ export const getSlotType = (slots: BuiltinSlot<string, string>[], slot: { name:
 
   return type;
 };
+
+export const VF_ENTITY_REGEXP = /{{\[(\w{1,32})]\.(\w{1,32})}}/gi;
+
+export interface JSONEntity {
+  startPos: number;
+  endPos: number;
+  entity: string;
+}
+export interface JSONUtterance {
+  text?: string;
+  entities?: JSONEntity[];
+}
+
+// extension of the String.prototype.replace format
+const continuousReplace = (text: string, regex: RegExp, replacer: (substring: string, ...args: any[]) => string): string => {
```

Comment:
@DecathectZero If we're on node 16 we should have [`String.prototype.replaceAll`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replaceAll)

Reply:
does `replaceAll` do it iteratively so that the `offset` parameter is relative to the previously replaced string?